### PR TITLE
Brain v2 fr

### PR DESCRIPTION
**Pull request by tlse-vins**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [x] If your changes affects code, did your write the tests?
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

Please provide a description of the change here. It's always best with screenshots, so don't hesitate to add some!

### CODE REVIEW

**Comment by tlse-vins** [comment]
Pour info dans mon codage il y a un espace avant le ? et ! mais ils n'y sont plus dans GitHub.
Est ce normal?

**Comment by VonOx** [comment]
Moi je les vois

File: server/config/brain/chat/questions.en.json [hunk 1]
```diff
@@ -1,12 +1,71 @@
...
-    "answers": ["Me too!", "Of course you love me..."]
+    "answers": ["Me too!", "I knew that ..."]
+  },
+  {
+    "label": "chat.joke",
+    "questions": ["Gladys tell me a joke.", "Gladys do you know a joke ?", "Tell me a joke."],
+    "answers": [
+      "Do you know the devil's joke? Too bad it's hell !",
+      "The other day I told a joke about my clothes. They were folded !",
+      "Who's in charge of the decoration at the Elysee? The Home Office.",
```
Comment (Pierre-Gilles) [inline server/config/brain/chat/questions.en.json]: Very french specific, doesn't work in english! You can remove it :)

File: server/config/brain/chat/questions.en.json [hunk 2]
```diff
@@ -1,12 +1,71 @@
...
+      "What do you say when calling a 4-headed monster? Hello, hello, hello, hello !",
+      "What do a postman and a knife thrower have in common? Both must have a lot of skill.",
```
Comment (Pierre-Gilles) [inline server/config/brain/chat/questions.en.json:28]: Does it work in english? I don't see the joke.

File: server/config/brain/chat/questions.en.json [hunk 3]
```diff
@@ -1,12 +1,71 @@
...
+      "This is the story of a bald man. Who has a hair on his tongue",
```
Comment (Pierre-Gilles) [inline server/config/brain/chat/questions.en.json]: Does it work in english? I don't see the joke.

File: server/config/brain/chat/questions.en.json [hunk 4]
```diff
@@ -1,12 +1,71 @@
...
+      "Why don't the police have acne? Because they have smooth skin !",
```
Comment (Pierre-Gilles) [inline server/config/brain/chat/questions.en.json]: Does it work in english? I don't see the joke.

File: server/config/brain/chat/questions.en.json [hunk 5]
```diff
@@ -1,12 +1,71 @@
...
+      "You have 2 chicks, you want one, what are you doing? You are pushing one !"
```
Comment (Pierre-Gilles) [inline server/config/brain/chat/questions.en.json]: Does it work in english? I don't see the joke.

File: server/config/brain/chat/questions.fr.json [hunk 6]
```diff
@@ -1,12 +1,75 @@
...
+      "Gladys, on fait un pile ou face ?",
+      "On joue à pile ou face ?",
+      "Pile ou face"
+    ],
+    "answers": ["Pile !", "Face !"]
+  },
+  {
+    "label": "chat.numbers0to10",
+    "questions": [
+      "Donne moi un chiffre entre 0 et 10",
```
Comment (Pierre-Gilles) [inline server/config/brain/chat/questions.fr.json]: If I ask Gladys "donne moi un chiffre entre 0 et 100", she replies with a number between 0 et 10.

Is there a way we could make this dynamic?

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]
Thanks for your PR!

I'm happy it finally worked on your side :) 

I added a few feedbacks. Be careful with your english.

For the English jokes, I would suggest taking native english jokes instead of trying to translate french jokes, because jokes are cultural and cannot be translated

File: server/config/brain/chat/questions.fr.json (hunk 6 above)
Comment (tlse-vins) [reply server/config/brain/chat/questions.fr.json]: I think we should use number extraction.
[(https://github.com/axa-group/nlp.js/blob/master/docs/v3/builtin-duckling.md#number-extraction)]
But I don't know how to do it.

Comment (Pierre-Gilles) [reply server/config/brain/chat/questions.fr.json]: Two solutions: 

- Remove the feature, as without this it'll lead to weird bugs. People don't know what is your implementation, so they'll just think Gladys is broken.

- Learn how to use this library and implement the number extraction :)

Comment (tlse-vins) [reply server/config/brain/chat/questions.fr.json]: I don't think that makes Gladys crash completely.
If you prefer that I delete this part while we set up the number extraction, I understand.
I also noticed that the NLP version was obsolete, would it be interesting to upgrade to version 4, and at the same time perform number extraction?

Comment (VonOx) [reply server/config/brain/chat/questions.fr.json]: This is a major version , there is certainly breaking change.

Time for another PR 😉

Comment (Pierre-Gilles) [reply server/config/brain/chat/questions.fr.json]: > If you prefer that I delete this part while we set up the number extraction, I understand. 

Yes please remove this part until the number extraction work. 

> I also noticed that the NLP version was obsolete

As Vonox said, it's not as simple as upgrading the version number, so yes: another PR.

File: server/config/brain/chat/questions.en.json [hunk 7]
```diff
@@ -1,12 +1,63 @@
...
     "label": "chat.i-love-you",
     "questions": ["I love you"],
-    "answers": ["Me too!", "Of course you love me..."]
+    "answers": ["Me too!", "I knew that ..."]
+  },
+  {
+    "label": "chat.joke",
+    "questions": ["Gladys tell me a joke.", "Gladys do you know a joke ?", "Tell me a joke."],
+    "answers": [
+      "Do you know the devil's joke? Too bad it's hell !",
```
Comment (thebradleysanders) [inline server/config/brain/chat/questions.en.json:25]: The English on these are great!

File: server/config/brain/chat/questions.en.json [hunk 8]
```diff
@@ -1,12 +1,63 @@
...
+      "The other day I told a joke about my clothes. They were folded !",
```
Comment (thebradleysanders) [inline server/config/brain/chat/questions.en.json]: Not sure if the English works on this one, or I don't understand the joke. All the others are great though.

File: server/config/brain/chat/questions.en.json [hunk 9]
```diff
@@ -1,12 +1,63 @@
...
+  {
+    "label": "chat.shifumi",
+    "questions": [
+      "Shall we make a shifumi ?",
+      "Gladys, are we playing shifumi ?",
+      "shifumi",
+      "shi fu mi",
+      "Gladys, are we doing a shi fu mi ?",
+      "Are we playing shifumi ?",
+      "Shall we make a stone, paper, scissors ?",
```
Comment (thebradleysanders) [inline server/config/brain/chat/questions.en.json]: Possible correction: "Should we play a rock, paper, scissors?"

File: server/config/brain/chat/questions.en.json [hunk 10]
```diff
@@ -1,12 +1,63 @@
...
+      "Gladys, are we playing rock, paper, scissors  ?",
+      "Are we playing rock, paper, scissors ?"
+    ],
+    "answers": ["Stone !", "Paper !", "Scissors !"]
+  },
+  {
+    "label": "chat.heads-or-tails",
+    "questions": [
+      "Shall we do a coin toss ?",
+      "Gladys, are we playing a coin toss ?",
```
Comment (thebradleysanders) [inline server/config/brain/chat/questions.en.json:57]: Possible Correction: "Gladys, are we playing coin toss ?"

File: server/config/brain/temperature-sensor/questions.en.json [hunk 11]
```diff
@@ -1,6 +1,11 @@
 [
   {
     "label": "temperature-sensor.get-in-room",
-    "questions": ["What's the temperature in the %room% ?", "Give me the temperature in the %room%"]
+    "questions": [
+      "What's the temperature in the %room% ?",
+      "Give me the temperature in the %room%",
+      "How much is it in the %room% ?",
```
Comment (thebradleysanders) [inline server/config/brain/temperature-sensor/questions.en.json]: Change to: "How cold is the %room% ?"
Add: "How hot is the %room% ?"

File: server/config/brain/chat/questions.en.json (hunk 2 above)
Comment (thebradleysanders) [reply server/config/brain/chat/questions.en.json:28]: I don't think so, as I don't see it either.

**Review by thebradleysanders — COMMENTED** [review]
Comments make on suggested changes in these files.

File: server/config/brain/chat/questions.en.json (hunk 9 above)
Comment (tlse-vins) [reply server/config/brain/chat/questions.en.json]: ok I will add.

File: server/config/brain/chat/questions.en.json (hunk 10 above)
Comment (tlse-vins) [reply server/config/brain/chat/questions.en.json:57]: ok I will add.

File: server/config/brain/temperature-sensor/questions.en.json (hunk 11 above)
Comment (tlse-vins) [reply server/config/brain/temperature-sensor/questions.en.json]: ok nice.

File: server/config/brain/chat/questions.en.json (hunk 8 above)
Comment (tlse-vins) [reply server/config/brain/chat/questions.en.json]: Je vais m'exprimer en Français, car je ne suis pas 100% à l'aise en anglais.
C'est un jeu de mot, ça fonctionne en français mais aussi en anglais. Vous n'allez pas être "plié de rire", ça c'est sure.
La plupart des blagues dans Google vous font rire? Il y a beaucoup de jeux de mot.
Je peux vous laisser la main afin de rajouter autant de blagues à mourir de rire que vous souhaitez.
C'est un premier ajout, qui sera enrichi en fonction des retours.

Comment (VonOx) [reply server/config/brain/chat/questions.en.json]: @tlse-vins Brad explique que la blague ne fonctionne pas en anglais  car "être plié de rire" est une expression française qui n'existe pas en Anglais. Tu devrai l'enlever ;) , je penses qu'il à raison.

@thebradleysanders thank you for feedback btw.

**Review by thebradleysanders — APPROVED** [review]
Perfect!

**Comment by Pierre-Gilles** [comment]
Merged in https://github.com/GladysAssistant/Gladys/commit/8f6bc72ad831aaf5af09e415f893b15268ec86fd

**Pull request closed without merging**